cmd/agent/app/start.go

```diff
@@ -261,19 +280,16 @@ func setupMetadataCollection(s *serializer.Serializer, hostname string) error {
 	} else {
 		log.Errorf("Unable to parse metadata_providers config: %v", err)
 	}
-	// Should be always true, except in some edge cases (multiple agents per host)
-	err = common.MetadataScheduler.AddCollector("host", hostMetadataCollectorInterval*time.Second)
-	if err != nil {
-		return log.Error("Host metadata is supposed to be always available in the catalog!")
-	}
-	err = common.MetadataScheduler.AddCollector("agent_checks", agentChecksMetadataCollectorInterval*time.Second)
-	if err != nil {
-		return log.Error("Agent Checks metadata is supposed to be always available in the catalog!")
-	}
-	if addDefaultResourcesCollector && runtime.GOOS == "linux" {
-		err = common.MetadataScheduler.AddCollector("resources", defaultResourcesMetadataCollectorInterval*time.Second)
+
+	// Adding default collectors if they were not listed in the configuration
+	for name, c := range defaultCollectors {
+		if c.os != "*" && runtime.GOOS != c.os {
+			continue
+		}
+		err = common.MetadataScheduler.AddCollector(name, c.interval*time.Second)
 		if err != nil {
-			log.Warn("Could not add resources metadata provider: ", err)
+			log.Error("Could not add resources metadata provider for %s: %v", name, err)
+			return err
```

Comment:
note: with your changes, if there's an error when adding the `resources` metadata collector, we now return an error instead of returning `nil`. This ultimately means that the Agent would shut down if the `resources` metadata can't be initialized.

Could you check the implementation and build tags of the `resources` collector to confirm this won't make the Agent shut down on startup in some scenarios/builds where we'd actually want the Agent to ignore the error and continue running?

Reply:
I replicated the behavior with the new implementation.